### PR TITLE
MOBC-471 || use values-in folder for Indonesian locale

### PR DESCRIPTION
**Pull request by scana**

### CODE REVIEW

File: plugin/build.gradle.kts [hunk 1]
```diff
@@ -75,6 +75,13 @@ dependencies {
...
+    testImplementation(gradleTestKit())
+    testImplementation("org.junit.jupiter:junit-jupiter:5.5.2")
+    testImplementation("org.assertj:assertj-core:3.11.1")
     testImplementation("com.squareup.okhttp3:mockwebserver:$okHttp")
 }
+
+
+tasks.withType<Test>().configureEach {
+    useJUnitPlatform()
+}
```
Comment (hbrylkowski) [inline plugin/build.gradle.kts]: there should be newline at the end of file

File: plugin/src/main/java/co/brainly/onesky/OneSkyPlugin.kt [hunk 2]
```diff
@@ -6,25 +6,27 @@ import co.brainly.onesky.task.UploadTranslationTask
...
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.jetbrains.annotations.TestOnly
 
 /**
  * @property apiKey API key found on OneSky account's settings page
  * @property apiSecret API secret found on OneSky account's settings page
  * @property projectId OneSky's project id for syncing files with
  * @property sourceStringFiles list of files to be synced with OneSky
+ * @property oneSkyApiUrl OneSky's API url, exposed only for tests, do not modify
```
Comment (hbrylkowski) [inline plugin/src/main/java/co/brainly/onesky/OneSkyPlugin.kt]: I think that modifying production class only for tests doesn't seem like a good idea. But injecting value on the other hand usually is. Also I could imagine that one sky could change their API url, or maybe they could have enterprise package own-hosted on custom domain. In general I would change comment to simply `@property oneSkyApiUrl OneSky's API url` and maybe carry over url somewhere to constant

File: plugin/src/main/java/co/brainly/onesky/task/DownloadTranslationsTask.kt [hunk 3]
```diff
@@ -101,6 +102,12 @@ open class DownloadTranslationsTask @Inject constructor(
                 val (locale, region) = code.split("-")
                 return "$locale-r$region"
             }
+
+            // https://developer.android.com/reference/java/util/Locale.html#toLanguageTag()
```
Comment (hbrylkowski) [inline plugin/src/main/java/co/brainly/onesky/task/DownloadTranslationsTask.kt:106]: then why not handle ` "iw", "ji",` ? Also maybe there is some more robust helper?

File: plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    projectId = 378760
+                    sourceStringFiles = listOf("strings.xml")
+                    oneSkyApiUrl = "${server.url("/")}"
+                }
+                  
+            """.trimIndent()
+        )
+
+        server.enqueueResponseWithJsonFile("project_languages_response_just_id.json")
+        server.enqueueResponseWithJsonFile("example_translation_file.xml")
```
Comment (hbrylkowski) [inline plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt]: Why json file has xml extension? Seems wrong

File: plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            "files not created in values-id directory"
+        )
+
+        assertEquals(
+            true,
+            moduleDir.resolve("src/main/res/values-in/strings.xml").exists(),
+            "files created in values-in directory"
+        )
+    }
+}
```
Comment (hbrylkowski) [inline plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt]: newline at the end of file

File: plugin/src/test/resources/project_languages_response_just_id.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "custom_locale": null,
+    "locale": "id",
+    "region": "",
+    "is_base_language": false,
+    "is_ready_to_publish": false,
+    "translation_progress": "99.8%",
+    "last_updated_at": "2020-11-09T03:34:42+0000",
+    "last_updated_at_timestamp": 1604892882
+  }]
+}
```
Comment (hbrylkowski) [inline plugin/src/test/resources/project_languages_response_just_id.json]: newline

File: plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+import co.brainly.onesky.util.settingsFile
+import okhttp3.mockwebserver.MockWebServer
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import org.junit.jupiter.api.Assertions.assertEquals as assertEquals
+
+class DownloadTranslationsTaskTest {
```
Comment (hbrylkowski) [inline plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt:13]: loose thoughts - you can test that here - in what seems to me like end to end test, but really it's just test of `androidDirName` getter. So test on it would seem sufficient. BUT since you have it here it will make any refactor failure proof and this is something I like. Maybe there is some place little deeper - that would make it possible to test it without calling HTTP API, and at the same time being refactor-proof. 

To summarize - looks good, but maybe there is a better level for this test.

File: plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt (hunk 4 above)
Comment (scana) [reply plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt]: Renamed method

File: plugin/src/main/java/co/brainly/onesky/OneSkyPlugin.kt (hunk 2 above)
Comment (scana) [reply plugin/src/main/java/co/brainly/onesky/OneSkyPlugin.kt]: Your comment got me thinking and I made overriding it more explicit and also annotated it properly to generate warnings for end-users.
Right now OneSky does not offer own-hosted instances (AFAIR), so I would say that it doesn't have to be exposed as a property.

File: plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt (hunk 7 above)
Comment (scana) [reply plugin/src/test/java/co/brainly/onesky/task/DownloadTranslationsTaskTest.kt:13]: Thank you, I really wanted to try out writing tests with GradleRunner, which is a recommended tool for writing Gradle plugins.
Will leave it for now as is, but this test should actually cover fully working example - I will treat this test as a good starting point for writing a test which fully covers responses from OneSky.

File: plugin/src/main/java/co/brainly/onesky/task/DownloadTranslationsTask.kt (hunk 3 above)
Comment (scana) [reply plugin/src/main/java/co/brainly/onesky/task/DownloadTranslationsTask.kt:106]: It's actually mashup of Android's way and Java's legacy approach. Haven't seen any handlers so far.
We do not support Hebrew and Yiddish (and I don't know how they are supported on Android) so I decided to handle it when somebody request for such a feature.

**Review by lekstel — APPROVED** [review]
